languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -10069,6 +10069,67 @@ though, and has since been slightly modified)-->
     </rulegroup>
   </category>
   <category id="CAT_GRAMMAIRE" name="Grammaire">
+      <rulegroup id="QUELLE_QUE" name="quelle que" default="temp_off">
+          <rule>
+              <pattern>
+                  <marker>
+                      <token regexp="yes">qu(&apostrophe;)?</token>
+                      <token>elle</token>
+                  </marker>
+                  <token regexp="yes">que|qu(&apostrophe;)</token>
+              </pattern>
+              <message>Une erreur semble être détectée dans cette phrase. Souhaitez-vous écrire <suggestion>quelle</suggestion> ?</message>
+              <example correction="quelle">Il le fera <marker>qu'elle</marker> que soit sa situation.</example>
+              <example correction="quelle">Il le fera <marker>qu'elle</marker> qu'en soit la raison.</example>
+          </rule>
+          <rule>
+              <pattern>
+                  <marker>
+                      <token regexp="yes">qu(&apostrophe;)?</token>
+                      <token>elles</token>
+                  </marker>
+                  <token regexp="yes">que|qu(&apostrophe;)</token>
+              </pattern>
+              <message>Une erreur semble être détectée dans cette phrase. Souhaitez-vous écrire <suggestion>quelles</suggestion> ?</message>
+              <example correction="quelles">Il le fera <marker>qu'elles</marker> qu'en soient les conséquences.</example>
+          </rule>
```

Comment:
This is a little tricky. Since your pattern only looks for `qu'elles qu'`, this means we would propose "quelles" even in this sentence: "On trouvera les responsables, qu'elles qu'ils soient" (resulting sentence: "On trouvera les responsables, quelles qu'ils soient").
There are different ways of solving this without restricting the rule too much. I'll leave it up to you to figure out a solution :)